api/campaigns/class-maybe-show-campaign.php

```diff
@@ -256,19 +257,28 @@ public function should_campaign_be_shown( $client_id, $campaign, $settings, $ref
 		}
 
 		// Handle segmentation.
-		$campaign_segment = isset( $settings->all_segments->{$campaign->s} ) ? $settings->all_segments->{$campaign->s} : false;
-		if ( ! empty( $campaign_segment ) ) {
+		$campaign_segment_ids = ! empty( $campaign->s ) ? explode( ',', $campaign->s ) : [];
+
+		if ( ! empty( $campaign_segment_ids ) ) {
+			$best_priority_segment_id = isset( $settings->best_priority_segment_id ) ?
```

Comment:
Why is the `best_priority_segment_id` set again here? It's already assigned on this line:

https://github.com/Automattic/newspack-popups/blob/88666acd412fa0e567846308f5bb094d06d1cab7/api/campaigns/class-maybe-show-campaign.php#L92

Reply:
This is a failsafe, mainly for testing purposes. I set it on line 92 to avoid running `get_best_priority_segment_id` for each prompt (instead, we run it once and cache the result in `$settings`). But if the property doesn't exist on `$settings` for whatever reason, we can still get it at this point.

Reply:
I think a comment should be added with an explanation, because looking at this class it's hard to imagine circumstances in which this property is not set.

Reply:
Added a comment explaining this in 378e969.